src/core/core_structs/device_model.jl

```diff
@@ -0,0 +1,7 @@
+abstract type AbstractDeviceFormulation end
+
+mutable struct DeviceModel{D <: PSY.PowerSystemDevice,
+                           B <: PSI.AbstractDeviceFormulation}
+    device::Type{D}
+    formulation::Type{B}
+end
```

Comment:
1) Why is this `mutable`?

2) Are you sure you want to be storing the type? You can probably get away with doing the following:

```
struct DeviceModel{Device <: PSY.PowerSystemDevice, Formulation <: PSI.AbstractDeviceFormulation} end
DeviceModel(D <: PSY.PowerSystemDevice, B <: PSI.AbstractDeviceFormulation) = Device{D, B}
```

Reply:
1. It is mutable such that the users can change it and request a rebuild if necessary. 

2. Yes, I want to store them. when calling build device I just want to call the `device_model`